iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/BooleanLiteralImpl.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep;
+
+import java.util.List;
+import org.sonar.iac.arm.tree.api.BooleanLiteral;
+import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
+import org.sonar.iac.arm.tree.impl.AbstractArmTreeImpl;
+import org.sonar.iac.common.api.tree.Tree;
+
+public class BooleanLiteralImpl extends AbstractArmTreeImpl implements BooleanLiteral {
+
+  private final SyntaxToken token;
+
+  public BooleanLiteralImpl(SyntaxToken token) {
+    this.token = token;
+  }
+
+  @Override
+  public boolean value() {
+    return "true".equals(token.value());
```

Comment:
Is it worth storing this value in a field, both here and in `NumericLiteralImpl`, so that it isn't recalculated on every access?

Reply:
That's a fair point.
Currently, we usually choose this basic approach: the method simply returns the value we are looking for, computing it every time.
I believe we do it for readability: it's easier to see what the method does instead of going through an additional layer of cache.
I mostly agree with this approach, as I don't expect a big win by caching the value: although it may happen that we call it a few times in our checks, I'm not expecting this to be more than 10 or at worst case 100 times, and the computation time wouldn't be that big.

This is also something you will find in the `children()` method that you can find in some tree elements, where we every time rebuild the list (See for example in the class HealthCheckInstructionImpl in the docker parser). We could theoretically replace it with a unique read-only List that would be computed once.